### ISSUE TITLE
Crash in Telemetry

### ISSUE THREAD

**Issue by iambmelt** [AzureAD/microsoft-authentication-library-for-android#769]
```java
09-26 10:01:05.114 5354-7474/com.microsoft.identity.client.sample.local E/AndroidRuntime: FATAL EXCEPTION: pool-5-thread-3
    Process: com.microsoft.identity.client.sample.local, PID: 5354
    java.util.ConcurrentModificationException
        at java.util.LinkedList$LinkIterator.remove(LinkedList.java:167)
        at com.microsoft.identity.common.internal.telemetry.Telemetry.flush(Telemetry.java:234)
        at com.microsoft.identity.common.internal.controllers.ApiDispatcher$1.run(ApiDispatcher.java:106)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
        at java.lang.Thread.run(Thread.java:818)
```

**Comment by shoatman** [comment]
@iambmelt - What were the repro steps for this issue?

**Comment by iambmelt** [comment]
@shoatman Clicked acquireToken in the sample app. The request failed, clicked acquireToken again, then crashed.

**Comment by shoatman** [comment]
@iambmelt - Can I get some additional detail?  For example:

- Device Type
- Broker installed?  Not installed?
- Which sample app?  (Test app or sample app?)
- If this is real bug and repros this consistently are we not saying that MSAL is broken?

**Comment by iambmelt** [comment]
- Nexus 6P
- `brokerHost` (test broker) installed
- Msal sample app

This appears to be an implementation problem -- from the [javadocs](https://developer.android.com/reference/java/util/LinkedList#top_of_page)

>The iterators returned by this class's iterator and listIterator methods are fail-fast: if the list is structurally modified at any time after the iterator is created, in any way except through the iterator's own remove or add methods, the iterator will throw a ConcurrentModificationException

From reading the `Telemetry` class, this is possible if `flush()` and `emit()` are called at the same time which is presumably what happened here

**Comment by Om83** [comment]
hi @shahzaibj  and @iambmelt , Can you please work together to resolve this soon . Appreciated!

**Comment by iambmelt** [comment]
ACK thanks @Om83

**Comment by shahzaibj** [comment]
@iambmelt Are you referring to the new java sample app that Dome created?

This one? https://github.com/Azure-Samples/ms-identity-android-java

**Comment by shahzaibj** [comment]
Fixed in common branch `release/1.0.6` by [Common PR 664](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/664)
